src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -420,18 +417,21 @@ private IEnumerable<IVsHierarchy> GetDeferredProjects()
         /// </summary>
         private bool DoesSolutionRequireAnInitialSaveAs()
         {
-            Debug.Assert(ThreadHelper.CheckAccess());
-
-            // Check if user is doing File - New File without saving the solution.
-            var value = GetVSSolutionProperty((int)(__VSPROPID.VSPROPID_IsSolutionSaveAsRequired));
-            if ((bool)value)
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
This pattern is more worrisome since now it waits to get a threadpool thread, block it, and tries to move to UI thread, and then execute, and release.

So where ever possible, make it async and let the caller do JTF.Run or whatever. And since this is private api, then we can easily maintain it either thow if not on ui thread and make sure all the callers calls it on UI or make this async...